setup/src/Setup/DrupalUtil.php

```diff
@@ -83,4 +85,41 @@ public static function guessSslParams(array $cmsDatabaseParams):array {
     return $ssl_params;
   }
 
+  /**
+   * Attempt to use a 'civicrm' labelled database connection if one exists.
+   *
+   * Otherwise default to using the same connection used by drupal.
+   * Also handle the special case where this is running as a test.
+   *
+   * @return string[]
+   *   An array of what database-config to use.
+   */
+  public static function get_database_configuration(): array {
+    if (drupal_valid_test_ua()) {
+      $config = Database::getConnectionInfo('civicrm_test');
```

Comment:
This might be very civi-jenkins-specific. I'm not sure yet if this would break some other drupal test environments.

Reply:
@demeritcowboy maybe but this is the same as in civicrm.install I'm not sure if civicarrot uses this?

Reply:
Ok I see what's happening. This is something MattG added for webform tests and it just [duplicates](https://github.com/civicrm/civicrm-drupal-8/blob/04a6d6a3f83aceb2ac2bff1494454364151ac5b9/tests/src/FunctionalJavascript/CiviCrmTestBase.php#L60) the "default" simpletest db settings. Carrot (or any test that runs using the drupal CiviCrmTestBase) does use it and in fact I doubt it's used in any of the jenkins runs (because I don't think even the drupal8 ones run using drupal's simpletest mechanism, so valid_ua would be false).

Anyway, long-winded way of saying "ok".